subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -931,7 +931,6 @@ private void recordNodeStarted(Node node) {
     }
 
     private void recordNodeCompleted(Node node) {
-        runningNodes.remove(node);
```

Comment:
What's the difference between `recordNodeCompleted()` and `nodeComplete()`? Can we clarify the difference a bit more by a different name maybe?

Reply:
`nodeCompleted == node finished executing` which means that the node also executed. `recordNodeCompleted` is more about doing the bookkeeping when the node moves to the completed state.

Reply:
I renamed `nodeCompleted` to `finishedExecuting`.